common/lib/xmodule/xmodule/modulestore/xml_importer.py

```diff
@@ -1167,5 +1168,5 @@ def _update_module_location(module, new_location):
     # explicitly set each field to its current value before triggering the save.
     if len(rekey_fields) > 0:
         for rekey_field_name in rekey_fields:
-            setattr(module, rekey_field_name, getattr(module, rekey_field_name))
+            module.fields[rekey_field_name].force_dirty(module)
```

Comment:
@cpennington `setattr` eventually triggers [`_mark_dirty`](https://github.com/edx/XBlock/blob/master/xblock/fields.py#L402-409) with value `EXPLICITLY_SET`. As you can see `_mark_dirty` does nothing if current field is already on dirty fields list.

The problem with mutable fields is that they are marked dirty [on read](https://github.com/edx/XBlock/blob/master/xblock/fields.py#L490-491). As a result, when we do `setattr(module, name, getattr(module, name))` with mutable fields we have the following:

* getattr goes to `xblock.fields.__get__` and [marks current value as dirty](https://github.com/edx/XBlock/blob/master/xblock/fields.py#L490-491)
* setattr goes to [`xblock.fields.__get__`](https://github.com/edx/XBlock/blob/master/xblock/fields.py#L504) and invokes mark_dirty
*  [`_mark_dirty`](https://github.com/edx/XBlock/blob/master/xblock/fields.py#L402-409) does nothing as it already contains current value
....
* ['_is_dirty'](https://github.com/edx/XBlock/blob/master/xblock/fields.py#L419-420) returns False as `baseline` is equal to current value.

Looks like it's an expected behavior: mutable fields should not be updated if their contents haven't been changed; mutable fields should be marked as dirty on read, as they can be changed without invoking set.